python/lsst/summit/utils/blockUtils.py

```diff
@@ -335,7 +344,12 @@ def getRows(self, block, seqNum=None):
         data : `pandas.DataFrame`
             The row data.
         """
-        rowsForBlock = self.data[self.data['blockNum'] == block]
+        # Because we query for a whole dayObs, but BLOCKs can overlap the day
+        # start/end, it's possible for the block's blockDayObs not to be the
+        # same as self.dayObs around the beginning or end of the day, so filter
+        # with an extra `& (self.data['blockDayObs'] == self.dayObs` when
+        # getting the relevant rows.
+        rowsForBlock = self.data[(self.data['blockNum'] == block) & (self.data['blockDayObs'] == self.dayObs)]
```

Comment:
It's more verbose, but I like using `numpy.logical_and` instead of `&` to avoid any chance of mixing up bitwise vs. logical in the rare scenarios where that matters.  But I think I may be in the minority.

Reply:
Yup, I like it - switched 🙂